### PR TITLE
CODEOWNERS: Drop file

### PR DESCRIPTION
**Pull request by russellb**
commit 7c6c49cc129b2b859cfb197351a7e0d51600dfde
Author: Russell Bryant <rbryant@redhat.com>
Date:   Fri Apr 19 08:57:53 2024 -0400

    CODEOWNERS: remove file
    
    I propose dropping the CODEOWNERS file.
    
    The use of CODEOWNERS prevents the default notifications
    setting of "Participating and @mentioned" from working. If you are in
    the CODEOWNERS, you get added to every PR and will get all
    notifications accordingly. This is pretty painful on a busy repo. If a
    maintainer desires that level of email notifications, they can change
    their setting to "All Activity" for the repository.
    
    CODEOWNERS is also not acting as access control to the repository.
    That access is managed via github teams, and those teams are granted
    different levels of access to the respository.
    
    In summary, by not using a top-level entry like this, I feel we give
    people more control over their notifications while not losing any
    level of access controls on the repo. It allows people to stick to a
    workflow that best works for them.
    
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>
    
See also https://github.com/instructlab/taxonomy/pull/718

### CODE REVIEW

**Comment by russellb** [comment]
My description as copied over from my taxonomy PR doesn't apply perfectly.

The file here was also used to tag people for PRs against specific files. I don't have a problem with that if those individuals really want it. I think it's a little simpler to just not use them, since the name of the file implies it's used for more than it is.

**Review by xukai92 — APPROVED** [review]
i also prefer to remove them.
it causes more confusion than helping us (at least for how it's set up now)

**Comment by nathan-weinberg** [comment]
> > If we get rid of it, I'm not opposed, but we need some other process for maintainers for triage PRs
> 
> Can you elaborate on this? Are you concerned about missing notifications? You can subscribe to all PRs if you want. You don't need CODEOWNERS for that.

What I mean is - will folks on the `CI Maintainers` GitHub Team simply be expected to keep up with PRs on their own? Since there wouldn't be notifications/automatic review additions via GitHub orchestrated by the `CODEOWNERS` file

**Comment by russellb** [comment]
> > > If we get rid of it, I'm not opposed, but we need some other process for maintainers for triage PRs
> > 
> > 
> > Can you elaborate on this? Are you concerned about missing notifications? You can subscribe to all PRs if you want. You don't need CODEOWNERS for that.
> 
> What I mean is - will folks on the `CI Maintainers` GitHub Team simply be expected to keep up with PRs on their own? Since there wouldn't be notifications/automatic review additions via GitHub orchestrated by the `CODEOWNERS` file

If you want to be notified of every PR, you can change your repo watch settings to "All Activity"

**Comment by nathan-weinberg** [comment]
> > > > If we get rid of it, I'm not opposed, but we need some other process for maintainers for triage PRs
> > > 
> > > 
> > > Can you elaborate on this? Are you concerned about missing notifications? You can subscribe to all PRs if you want. You don't need CODEOWNERS for that.
> > 
> > 
> > What I mean is - will folks on the `CI Maintainers` GitHub Team simply be expected to keep up with PRs on their own? Since there wouldn't be notifications/automatic review additions via GitHub orchestrated by the `CODEOWNERS` file
> 
> If you want to be notified of every PR, you can change your repo watch settings to "All Activity"

I guess it seems to make more sense to me for folks who are officially supposed to be looking at these PRs to get notified about them and opt-out rather than vice-versa, but I don't feel strongly one way or the other. Personally I do a lot of filtering on my email side to organize things versus having to go and check the Pull Requests myself and opt-into them.

**Comment by russellb** [comment]
new corresponding PR for the taxonomy repo #741

**Comment by russellb** [comment]
> > > > > If we get rid of it, I'm not opposed, but we need some other process for maintainers for triage PRs
> > > > 
> > > > 
> > > > Can you elaborate on this? Are you concerned about missing notifications? You can subscribe to all PRs if you want. You don't need CODEOWNERS for that.
> > > 
> > > 
> > > What I mean is - will folks on the `CI Maintainers` GitHub Team simply be expected to keep up with PRs on their own? Since there wouldn't be notifications/automatic review additions via GitHub orchestrated by the `CODEOWNERS` file
> > 
> > 
> > If you want to be notified of every PR, you can change your repo watch settings to "All Activity"
> 
> I guess it seems to make more sense to me for folks who are officially supposed to be looking at these PRs to get notified about them and opt-out rather than vice-versa, but I don't feel strongly one way or the other. Personally I do a lot of filtering on my email side to organize things versus having to go and check the Pull Requests myself and opt-into them.

I agree that people responsible for looking should be looking, but I propose letting people accomplish that goal using the workflow that best works for them. It may include "subscribe to all github notifications", but it doesn't have to. There are other ways to query for PRs across repos that you haven't touched yet.

**Comment by nathan-weinberg** [comment]
Will let others weigh-in - I am fine either way :+1:

**Review by bjhargrave — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]
I agree about removing it also as the value on top of the GH teams hasn't been of use to the project
